sympy/calculus/tests/test_util.py

```diff
@@ -63,6 +63,55 @@ def test_not_empty_in():
         Union(Interval(-2, -1, True, False), Interval(2, oo))
 
 
+def test_periodicity():
+    x = Symbol('x')
+    y = Symbol('y')
+
+    assert periodicity(sin(2*x), x) == pi
+    assert periodicity((-2)*tan(4*x), x) == pi/4
+    assert periodicity(sin(x)**2, x) == 2*pi
```

Comment:
> ```python
> In [3]: periodicity(sin(x)**y, x)
> Out[3]: 2⋅π
> ```

This is periodic in `x` with a period same as that of `sin(x)`. 
Hence, `2*pi` is the answer.
I think this case is similar to this [testcase](https://github.com/sympy/sympy/pull/11277/files#diff-c637a1ca6aa8a6dd4cd50503f1ccd0fcR87).